File: app/src/main/java/me/rakshakhegde/dagger_android_injector/screens/main_screen/MainViewModel.java
package me.rakshakhegde.dagger_android_injector.screens.main_screen;

import javax.inject.Inject;

import me.rakshakhegde.dagger_android_injector.R;

/**
 * Created by rakshakhegde on 26/04/17.
 */

public class MainViewModel {

	private MainView V;

	@Inject
	MainViewModel(MainView V) {
		this.V = V;
	}

	public void showBtnClicked() {
		V.showMessage(R.string.showing_toast);
	}
}

File: app/src/main/java/me/rakshakhegde/dagger_android_injector/screens/main_screen/MainViewModel.kt
package me.rakshakhegde.dagger_android_injector.screens.main_screen

import me.rakshakhegde.dagger_android_injector.R
import javax.inject.Inject

/**
 * Created by rakshakhegde on 26/04/17.
 */

class MainViewModel @Inject constructor(private val V: MainView) {

	fun showBtnClicked() = V.showMessage(R.string.showing_toast)
}
